commit 96ea45d1938504550bba7ec1349622db8a569b19
Author: Michael McCandless <mikemccand@apache.org>
Date:   Sat Jul 7 12:28:04 2007 +0000

    LUCENE-952: force synchronized access to writer instance variable to fix infinite spin loop in TestGdataIndexer
    
    git-svn-id: https://svn.apache.org/repos/asf/lucene/java/trunk@554189 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/contrib/gdata-server/src/core/src/java/org/apache/lucene/gdata/search/index/GDataIndexer.java b/contrib/gdata-server/src/core/src/java/org/apache/lucene/gdata/search/index/GDataIndexer.java
index 7cb8885..68369de 100755
--- a/contrib/gdata-server/src/core/src/java/org/apache/lucene/gdata/search/index/GDataIndexer.java
+++ b/contrib/gdata-server/src/core/src/java/org/apache/lucene/gdata/search/index/GDataIndexer.java
@@ -400,6 +400,11 @@ public class GDataIndexer {
 
     }
 
+    // Used only for testing
+    protected synchronized IndexWriter getWriter() {
+      return this.writer;
+    }
+
     /**
      * This factory method creates a new GDataIndexer using a instance of
      * {@link IndexTask}
diff --git a/contrib/gdata-server/src/core/src/test/org/apache/lucene/gdata/search/index/TestGdataIndexer.java b/contrib/gdata-server/src/core/src/test/org/apache/lucene/gdata/search/index/TestGdataIndexer.java
index 87e16b7..1b09dc0 100755
--- a/contrib/gdata-server/src/core/src/test/org/apache/lucene/gdata/search/index/TestGdataIndexer.java
+++ b/contrib/gdata-server/src/core/src/test/org/apache/lucene/gdata/search/index/TestGdataIndexer.java
@@ -552,7 +552,7 @@ public void testAddDocument() throws IOException {
         latch.countDown();
         documentLatch.await(5000,TimeUnit.MILLISECONDS);
         // wait active for the commit
-        while(this.indexer.writer != null){}
+        while(this.indexer.getWriter() != null){}
         
         IndexSearcher s = new IndexSearcher(this.dir);
         Hits h = s.search(new TermQuery(delTerm));
